src/QueryType/Update/RequestBuilder.php

```diff
@@ -181,6 +184,30 @@ public function buildRollbackXml(): string
         return '<rollback/>';
     }
 
+    /**
+     * Build XML for a raw command.
+     *
+     * @param RawXML $command
+     *
+     * @return string
+     */
+    public function buildRawXmlXml(RawXML $command): string
+    {
+        $xml = '';
+
+        foreach ($command->getCommands() as $raw) {
+            // unwrap grouped commands, they must be consolidated in a single <update>
+            if (false !== ($pos = strpos($raw, '<update'))) {
```

Comment:
> Doesn't it make sense to immediately "unwrap" the command in RawXml::addCommand()?

That would mean, from a user's perspective, that `getCommands` doesn't return the 'same' commands that were added with `addCommand` or `addCommands`. I think it would be fair to expect `assertSame` to work on add/get.

Furthermore, I'm already thinking about how to go about raw JSON commands. The way we build requests, it will be straightforward to generate XML from the decoded JSON string. I wouldn't do that in `addCommand` because a user will definitely expect JSON back from `getCommands`. It would break the symmetry if you could get your verbatim JSON string back but not your verbatim XML string.

> It could be done using a simple multi line preg_replace().

I tried and failed. My current code works for multiple comments and/or PIs before *and* after the root element, with or without whitespace before, after and inbetween those. It also works for whitespace and attributes in the opening tag. (Attributes not throwing it off is incidental. There are no documented attributes, and Solr happily ignores them too.) I couldn't get that to work for every valid combination if I wanted to make sure only a root `<update>` element gets picked up. If we can safely assume there will never ever be a nested `<update>` element anywhere in a valid XML command, it could be done with little effort. Do we feel confident about that assumption?